### PR TITLE
Cmake platforms

### PR DESCRIPTION
**Pull request by wsnipex**
simplify dependency handling/listing

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -154,37 +156,19 @@ set(dyload_optional Bluray
...
-endforeach()
-
-foreach(dep ${required_deps})
-  core_require_dep(${dep})
-endforeach()
-
-foreach(dep ${optional_deps})
-  core_optional_dep(${dep})
-endforeach()
+core_require_dep("${required_deps}")
```
Comment (fetzerch) [inline CMakeLists.txt]: if you want to get rid of the quotes here, you could also define the function without parameter and loop over ARGN

File: cmake/scripts/common/Macros.cmake [hunk 2]
```diff
@@ -327,31 +327,37 @@ endmacro()
...
 #   dependency added to ${SYSTEM_INCLUDES}, ${DEPLIBS} and ${DEP_DEFINES}
-function(core_require_dep dep)
-  find_package(${dep} REQUIRED)
-  string(TOUPPER ${dep} depup)
-  list(APPEND SYSTEM_INCLUDES ${${depup}_INCLUDE_DIRS})
-  list(APPEND DEPLIBS ${${depup}_LIBRARIES})
-  list(APPEND DEP_DEFINES ${${depup}_DEFINITIONS})
-  export_dep()
+function(core_require_dep dep_list)
+  foreach(dep ${dep_list})
```
Comment (fetzerch) [inline cmake/scripts/common/Macros.cmake]: you could use `foreach(dep IN LISTS dep_list)` here.

File: cmake/scripts/linux/platform/X11.cmake [hunk 3]
```diff
@@ -1,8 +1,3 @@
-core_require_dep(OpenGl)
-core_require_dep(EGL)
-core_require_dep(X)
-core_require_dep(XRandR)
-core_require_dep(LibDRM)
-core_optional_dep(VAAPI)
-core_optional_dep(VDPAU)
-core_optional_dep(OpenGLES)
+set(ENABLE_X11 ON)
```
Comment (fetzerch) [inline cmake/scripts/linux/platform/X11.cmake]: wouldn't it be better to change the relevant places to use the variable X_FOUND?

**Review by fetzerch — COMMENTED** [review]
minors only

**Comment by wsnipex** [comment]
thanks, updated

**Review by fetzerch — APPROVED** [review]

**Comment by lrusak** [comment]
This looks good to me!

**Pull request merged**